class.photon.php

```diff
@@ -65,6 +65,99 @@ private function setup() {
 
 		// Helpers for maniuplated images
 		add_action( 'wp_enqueue_scripts', array( $this, 'action_wp_enqueue_scripts' ), 9 );
+
+		/**
+		 * Allow Photon to disable uploaded images resizing and use its own resize capabilities instead.
+		 *
+		 * @module photon
+		 *
+		 * @since 7.1.0
+		 *
+		 * @param bool false Should Photon enable noresize mode. Default to false.
+		 */
+		if ( apply_filters( 'jetpack_photon_noresize_mode', false ) ) {
+			$this->enable_noresize_mode();
+		}
+	}
+
+	/**
+	 * Enables the noresize mode for Photon, allowing to avoid intermediate size files generation.
+	 */
+	private function enable_noresize_mode() {
+		// The main objective of noresize mode is to disable additional resized image versions creation.
+		// This filter handles removal of additional sizes.
+		add_filter( 'intermediate_image_sizes_advanced', array( __CLASS__, 'filter_photon_noresize_intermediate_sizes' ) );
+
+		// This allows to assign the Photon domain to images that normally use the home URL as base.
+		add_filter( 'jetpack_photon_domain', array( __CLASS__, 'filter_photon_norezise_domain' ), 10, 2 );
+
+		add_filter( 'the_content', array( __CLASS__, 'filter_content_add' ), 0 );
+
+		// Jetpack hooks in at six nines (999999) so this filter does at seven.
+		add_filter( 'the_content', array( __CLASS__, 'filter_content_remove' ), 9999999 );
+
+		// Regular Photon operation mode filter doesn't run when is_admin(), so we need an additional filter.
+		// This is temporary until Jetpack allows more easily running these filters for is_admin().
+		if ( is_admin() ) {
+			add_filter( 'image_downsize', array( $this, 'filter_image_downsize' ), 5, 3 );
+			add_filter( 'jetpack_photon_admin_allow_image_downsize', array( __CLASS__, 'filter_photon_noresize_allow_downsize' ), 10, 2 );
```

Comment:
What's the benefit of calling another function here instead of doing `__return_true`?